src/include/btree.i

```diff
@@ -981,8 +981,7 @@ __wt_page_can_evict(WT_SESSION_IMPL *session, WT_PAGE *page, int check_splits)
 	 * been written in the checkpoint, leaving the checkpoint inconsistent.
 	 */
 	if (btree->checkpointing &&
-	    (__wt_page_is_modified(page) ||
-	    F_ISSET(mod, WT_PM_REC_MULTIBLOCK))) {
+	    (__wt_page_is_modified(page) || F_ISSET(mod, WT_PM_REC_MASK))) {
```

Comment:
@michaelcahill, this change bothers me, so I wanted to write up what I'm thinking and talk it over with you when you come online.

Primarly, I'm worried about performance: it seems to me this change will mean we can't evict a page that's ever been dirtied, during checkpoint. I think that's a serious change with respect to performance.

I've been trying to think through the path where it's a problem for the tree to be marked dirty during a checkpoint.

Obviously, if we don't mark the tree clean, we may do an additional checkpoint on close that's not necessary, but is there a more serious issue?

Listing the steps for a multi-block split and checkpoint race:
1. Write a leaf page, set `WT_PM_REC_MULTIBLOCK`, mark the page clean
2. Start a checkpoint
2. Evict the leaf page, splitting the multiple blocks into the parent

If checkpoint writes the leaf page's parent before step 3, we'll merge the multiple blocks from the leaf page into the parent page.

If we write the leaf page's parent after step 3, the result should be identical: we've just replaced a merge step in the parent page's reconciliation into a sequential walk across the identical list of block addresses.

In other words, other than marking the tree dirty, we should get identical results no matter in what order we doing the eviction with respect to the checkpoint.

I'm less happy with the idea of deepening the tree during a checkpoint: I can't think of any reason it wouldn't work, but it's a heavy-weight operation and scares me a little.

All that said, I guess I still think split should always mark the tree dirty because I don't see an easy way to avoid it consistently.

Note the existing split code might have marked a page/tree dirty even before the change you made in this branch: when doing a `split-rewrite` or `split-multi` we jump into the col/row-store modify code to apply updates we couldn't write, and that can mark the page/tree dirty.

So, I think we should be consistent and the split code should always mark pages/trees dirty, which means we can't go into the split code during a checkpoint if we want to ensure a tree "cleaned" by checkpoint remains clean.

First question: maybe we should get rid of the notion of trying to keep the tree clean after a checkpoint? We could revert the split code to not mark the tree dirty in most cases, and just not worry about the additional checkpoints on close?

Second question: would it be correct to assert in the "mark the page/tree dirty" functions that we're not currently checkpointing the file, would that help debug this?

Third question: the previous check for `WT_PM_REC_MULTIBLOCK`, makes us safe with respect to `split-rewrite` and `split-multi`, but what if we called `split-insert`, which can also mark the tree dirty? Currently, `split-insert` ignores clean pages, so I think we're OK, but the fact the tests are widely separated worries me a little.

Fourth question: to wrap up, I don't see the reason for this change, and I think it will cost us performance.